types/webpack-dev-server/index.d.ts

```diff
@@ -133,93 +198,35 @@ declare namespace WebpackDevServer {
-         * devServer.injectHot to true will result in always injecting. It is
-         * possible to provide a function to inject conditionally
-         */
-        injectHot?: boolean | ((compilerConfig: webpack.Compiler) => boolean) | undefined;
-        /**
-         * Toggle between the dev-server's two different modes.  By default the
-         * application will be served with inline mode enabled.  This means
-         * that a script will be inserted in your bundle to take care of live
-         * reloading, and build messages will appear in the browser console.
-         */
-        inline?: boolean | undefined;
-        /**
-         * When lazy is enabled, the dev-server will only compile the bundle
-         * when it gets requested.  This means that webpack will not watch any
-         * file changes.
-         */
-        lazy?: boolean | undefined;
+        /* The Unix socket to listen to (instead of a host). */
+        ipc?: boolean | string | undefined;
         /**
          * By default, the dev-server will reload/refresh the page when file
          * changes are detected. devServer.hot option must be disabled or
          * devServer.watchContentBase option must be enabled in order for
-         * liveReload to take effect. Disable devServer.liveReload by setting
-         * it to false
+         * liveReload to take effect. Disable devServer.liveReload by setting it
+         * to false
          */
         liveReload?: boolean | undefined;
         /**
-         * The object is passed to the underlying webpack-dev-middleware. See
-         * [documentation](https://github.com/webpack/webpack-dev-middleware#mimetypes)
-         * for usage notes.
+         * Provides the ability to execute custom middleware after all other
+         * middleware internally within the server.
          */
-        mimeTypes?:
-            | {
-                  [key: string]: string[];
-              }
-            | {
-                  typeMap?: {
-                      [key: string]: string[];
-                  } & {
-                      force: boolean;
-                  } | undefined;
-              } | undefined;
+        onAfterSetupMiddleware?: ((server: WebpackDevServer) => void) | undefined;
         /**
-         * With noInfo enabled, messages like the webpack bundle information
-         * that is shown when starting up and after each save,will be hidden.
-         * Errors and warnings will still be shown.
+         * Provides the ability to execute custom middleware prior to all other
+         * middleware internally within the server.
          */
-        noInfo?: boolean | undefined;
+        onBeforeSetupMiddleware?: ((server: WebpackDevServer) => void) | undefined;
         /**
          * Provides an option to execute a custom function when
          * webpack-dev-server starts listening for connections on a port.
          */
         onListening?: ((server: WebpackDevServer) => void) | undefined;
```

Comment:
```suggestion
        onListening?: ((devServer: WebpackDevServer) => void) | undefined;
```
can we change this as well? thx!

Reply:
Of course, thanks for your suggestions!